project_1/include/utils.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <cstdint>
+#include <iostream>
+#include <chrono>
+
+std::uint64_t GetCurrentTimeInMicroseconds();
+
+enum event_type: std::uint8_t {
+    SEND_DATA,
+    ACKNOWLEDGEMENT,
+};
+
+// Type for server receiver, contains only ID
+struct ServerBase {
+protected:
+    std::uint64_t id;
+public:
+    ServerBase(std::uint32_t);
+    ServerBase() = default;
+    ~ServerBase() = default;
+
+    std::uint32_t GetID() const;
+};
+
+// Type for sending server, contains ID and distance to receiver
+struct ServerSender : ServerBase {
+private:
+    std::uint32_t distance_μs;
+public:
+    ServerSender(std::uint32_t, std::uint32_t);
+    ServerSender() = default;
+    ~ServerSender() = default;
+
+    std::uint32_t GetDistance() const;
+};
+
+// Type for package
+struct PackageHeader {
+private:
+    std::uint64_t sender_id;
+    std::uint64_t receiver_id;
+    std::uint64_t sending_time;
+    std::uint32_t package_index;
+    std::uint32_t RTT;
```

Comment:
This might be all right but it it worries me a bit that the packet knows about RTT. The RTT can only become known when the acknowledgement arrives back to the sender, after the packet is received.

Reply:
I agree, packet doesn't know anything about RTT. To order packets we need to calculate approximate travel time, I should use it instead of RTT.